tensorboard/components_polymer3/tf_dashboard_common/data-loader-behavior.ts

```diff
@@ -66,28 +83,11 @@ export function DataLoaderBehavior<Item, Data>(
      * This function will be called when a response from a request to that
      * data URL is successfully received.
      */
-    loadDataCallback!: (component: this, datum: Item, data: Data) => void;
-
-    public requestManager!: RequestManager;
-
-    // A function that takes a datum as argument and makes the HTTP
-    // request to fetch the data associated with the datum. It should return
-    // a promise that either fullfills with the data or rejects with an error.
-    // If the function doesn't bind 'this', then it will reference the element
-    // that includes this behavior.
-    // The default implementation calls this.requestManager.request with
-    // the value returned by this.getDataLoadUrl(datum) (see below).
-    // The only place getDataLoadUrl() is called is in the default
-    // implementation of this method. So if you override this method with
-    // an implementation that doesn't call getDataLoadUrl, it need not be
-    // provided.
-    requestData = (datum: Item) => {
-      return this.requestManager.request(this.getDataLoadUrl(datum));
-    };
-
-    // A function that takes a datum and returns a string URL for fetching
-    // data.
-    getDataLoadUrl!: (datum: Item) => string;
+    loadDataCallback!: (component: this, item: Item, data: Data) => void;
```

Comment:
Do you think there is a benefit in "batching" this API, too? i.e., call the callback once per all `items`.

Reply:
Ah, I think I made a mistake. I definitely meant to only commit the changes on `onLoadFinish`. I will send a separate PR shortly.

Reply:
Forgot to comment here: I did play around with this and everything still
works, and I see a significant improvement in paint time.

It’d be slightly easier for me if I could send that PR after these two
land, just so that I don’t have to futz with conflicts; is that okay
with you?

Reply:
Awesome. It is awesome that it improves the paint time, too.

Please feel free to work on it.